code/datums/components/butchering.dm

```diff
@@ -99,50 +99,75 @@
+				results += new remains (location)
+			results += new remains (location)
 
-		meat.butcher_results.Remove(bones) //in case you want to, say, have it drop its results on gib
+		target.butcher_results.Remove(remains) //in case you want to, say, have it drop its results on gib
 
-	for(var/V in meat.guaranteed_butcher_results)
-		var/obj/sinew = V
-		var/amount = meat.guaranteed_butcher_results[sinew]
+	for(var/guaranteed_result_typepath in target.guaranteed_butcher_results)
+		var/obj/guaranteed_remains = guaranteed_result_typepath
+		var/amount = target.guaranteed_butcher_results[guaranteed_remains]
 		for(var/i in 1 to amount)
-			results += new sinew (T)
-		meat.guaranteed_butcher_results.Remove(sinew)
+			results += new guaranteed_remains (location)
+		target.guaranteed_butcher_results.Remove(guaranteed_remains)
 
 	for(var/obj/item/carrion in results)
 		var/list/meat_mats = carrion.has_material_type(/datum/material/meat)
 		if(!length(meat_mats))
 			continue
-		carrion.set_custom_materials((carrion.custom_materials - meat_mats) + list(GET_MATERIAL_REF(/datum/material/meat/mob_meat, meat) = counterlist_sum(meat_mats)))
+		carrion.set_custom_materials((carrion.custom_materials - meat_mats) + list(GET_MATERIAL_REF(/datum/material/meat/mob_meat, target) = counterlist_sum(meat_mats)))
+
+	// transfer delicious reagents to meat
+	if(target.reagents)
+		var/meat_produced = 0
+		for(var/obj/item/food/meat/slab/target_meat in results)
+			meat_produced += 1
+		for(var/obj/item/food/meat/slab/target_meat in results)
+			target.reagents.trans_to(target_meat, target.reagents.total_volume / meat_produced, remove_blacklisted = TRUE)
+
+	// dont forget yummy diseases either!
+	if(iscarbon(target))
+		var/mob/living/carbon/host_target = target
+		var/list/diseases = host_target.get_static_viruses()
+		if(LAZYLEN(diseases))
+			var/list/datum/disease/diseases_to_add = list()
+			for(var/datum/disease/disease as anything in diseases)
+				// admin or special viruses that should not be reproduced
+				if(disease.spread_flags & (DISEASE_SPREAD_SPECIAL | DISEASE_SPREAD_NON_CONTAGIOUS))
+					continue
+
+				diseases_to_add += disease
+			if(LAZYLEN(diseases_to_add))
+				for(var/obj/diseased_remains in results)
+					diseased_remains.AddComponent(/datum/component/infective, diseases_to_add)
 
 	if(butcher)
-		butcher.visible_message(span_notice("[butcher] butchers [meat]."), \
-								span_notice("You butcher [meat]."))
-	butcher_callback?.Invoke(butcher, meat)
-	meat.harvest(butcher)
-	meat.log_message("has been butchered by [key_name(butcher)]", LOG_ATTACK)
-	meat.gib(FALSE, FALSE, TRUE)
+		butcher.visible_message(span_notice("[butcher] butchers [target]."), \
+								span_notice("You butcher [target]."))
+	butcher_callback?.Invoke(butcher, target)
+	target.harvest(butcher)
+	target.log_message("has been butchered by [key_name(butcher)]", LOG_ATTACK)
+	target.gib(FALSE, FALSE, TRUE)
```

Comment:
```suggestion
	target.gib(no_bodyparts = TRUE)
```

Reply:
Tested and it works.

Reply:
Apparently the linter isn't too happy about this...

Reply:
Ah I see unfortunately that you have encountered the curse of "people were just entering three unlabelled boolean in a row instead of using kwargs and consequently a bunch of overrides of `gib()` do not have the args specified" 😔

Reply:
It'd honestly be better to use a bitflag instead of multiple boolean positional args or kwargs.